### PR TITLE
Let the title, content and excerpt be empty

### LINKED ISSUE #745: Can't update the post with empty content, title or excerpt
I know you can't update a post without the content, title AND excerpt empty, but you can leave two fields blank if one is filled. The API fails to do that.

### PR DESCRIPTION
**Pull request by iseulde**
Fixes #745. Brings up another issue with the excerpt...

### CODE REVIEW

File: lib/class-wp-json-base-posts-controller.php [hunk 1]
```diff
@@ -68,6 +70,8 @@ public function get_item( $request ) {
 			return new WP_Error( 'json_post_invalid_id', __( 'Invalid post ID.' ), array( 'status' => 404 ) );
 		}
 
+		setup_postdata( $post );
```
Comment (rmccue) [inline lib/class-wp-json-base-posts-controller.php]: If we set this up here, we need to ensure we restore it afterwards. We're already doing that inside `prepare_item_for_response`, do we need it here too?

File: lib/class-wp-json-base-posts-controller.php [hunk 2]
```diff
@@ -375,31 +379,28 @@ protected function prepare_item_for_database( $request ) {
 		}
 
 		// Post title
-		if ( ! empty( $request['title'] ) ) {
+		if ( isset( $request['title'] ) ) {
 			if ( is_string( $request['title'] ) ) {
 				$prepared_post->post_title = wp_kses_post( $request['title'] );
-			}
-			elseif ( ! empty( $request['title']['raw'] ) ) {
+			} else if ( isset( $request['title']['raw'] ) ) {
```
Comment (rmccue) [inline lib/class-wp-json-base-posts-controller.php]: Should be `elseif` per coding standards.

File: tests/test-json-posts-controller.php [hunk 3]
```diff
@@ -786,7 +828,11 @@ protected function check_get_posts_response( $response, $context = 'view' ) {
 
 		$all_data = $response->get_data();
 		$data = $all_data[0];
+
 		$post = get_post( $data['id'] );
+
+		setup_postdata( $post );
```
Comment (rmccue) [inline tests/test-json-posts-controller.php]: Is there a reason to do this inside the tests?

**Comment by rmccue** [comment]
Will need a merge resolution too. :)

**Comment by danielbachhuber** [comment]
@iseulde I think this is addressed by #953 and #952. Can you confirm?

**Comment by danielbachhuber** [comment]
This has been addressed a while back.

**Pull request closed without merging**